var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,176 @@
+
+        # Currently FFTW_HOME and LIBXML2_HOME are used by CMake.
+        # Any CMake warnings about other variables are benign.
+        xml2_prefix = self.spec['libxml2'].prefix
+        args.append('-DLIBXML2_HOME={0}'.format(xml2_prefix))
+        args.append('-DLibxml2_INCLUDE_DIRS={0}'.format(xml2_prefix.include))
+        args.append('-DLibxml2_LIBRARY_DIRS={0}'.format(xml2_prefix.lib))
+
+        fftw_prefix = self.spec['fftw'].prefix
+        args.append('-DFFTW_HOME={0}'.format(fftw_prefix))
+        args.append('-DFFTW_INCLUDE_DIRS={0}'.format(fftw_prefix.include))
+        args.append('-DFFTW_LIBRARY_DIRS={0}'.format(fftw_prefix.lib))
+
+        args.append('-DBOOST_ROOT={0}'.format(self.spec['boost'].prefix))
+        args.append('-DHDF5_ROOT={0}'.format(self.spec['hdf5'].prefix))
+
+        # Default is MPI, serial version is convenient for cases, e.g. laptops
+        if '~mpi' in self.spec:
+            args.append('-DQMC_MPI=0')
+
+        # When '-DQMC_CUDA=1', CMake automatically sets:
+        # '-DQMC_MIXED_PRECISION=1'
+        #
+        # Note: there is a double-precision CUDA path, but it is deprecated
+        if '+cuda' in self.spec:
+            args.append('-DQMC_CUDA=1')
+
+        # this is for the experimental mixed-prescision CPU code
+        if '+mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=1')
+
+        return args
+
+    def install(self, spec, prefix):
+        """Make the install targets"""
+
+        # QMCPACK 'make install' does nothing, which causes
+        # Spack to throw an error.
+        #
+        # This install method creates the top level directory
+        # and copies the bin subdirectory into the appropriate
+        # location. We do not copy include or lib at this time due
+        # to technical difficulties in qmcpack itself.
+
+        mkdirp(prefix)
+
+        # We assume cwd is self.stage.source_path
+
+        # install manual
+        install_tree('manual', prefix.manual)
+
+        # install nexus
+        install_tree('nexus', prefix.nexus)
+
+        with working_dir(self.build_directory):
+            # QMCPACK 'make install' does nothing, which causes
+            # Spack to throw an error.
+            #
+            # This install method creates the top level directory
+            # and copies the bin subdirectory into the appropriate
+            # location. We do not copy include or lib at this time due
+            # to technical difficulties in qmcpack itself.
+            mkdirp(prefix)
+
+            # install binaries
+            install_tree('bin', prefix.bin)
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check_build(self):
+        """Run ctest after building binary.
+        Use 'spack install --run-tests qmcpack'.
+        It can take 24 hours to run all the regression tests
+        and unit tests. """
+        self.ctest
+        pass
```

Comment:
No need for `pass` here. I think this would look better as `ctest()` than `self.ctest`.

Reply:
@adamjstewart I have been looking at this more carefully and it does not look like the arguments are being passed correctly to `ctest`. Possibly this could be a white space issue? Unfortunately, all of the command-line arguments for ctest are of the form:
`ctest -<flag> <name>`

I am about to create a GitHub issue on this... unless you are certain that this is working properly